Fix part of #10306: Added async keyword to asynchronous functions that return a promise

shubhamkorde opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #10306.
2. This PR does the following: It replaces the getPermissions() function with async getPermissionsAsync() as per the issue requirement

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes made in this PR work correctly.
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Oppiabot can assign anyone for review/help if you leave a comment like the following: "{{Question/comment}} @{{reviewer_username}} PTAL"
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

shubhamkorde committed:
Fix part of #10306: Added async keyword to asynchronous functions that return a promise

shubhamkorde committed:
Fix part of #10306: Added async keyword to asynchronous functions that return a promise

shubhamkorde committed:
Fix part of #10306: Added async keyword to asynchronous functions that return a promise

shubhamkorde commented:
@kevintab95 @nithusha21 PTAL!

shubhamkorde committed:
updated changes from develop to fix failing issues

shubhamkorde committed:
Merge pull request #1 from oppia/develop

Update latest content

shubhamkorde committed:
Merge branch 'async-functions-update' of https://github.com/shubhamkorde/oppia into async-functions-update

prayutsu committed:
Fix part of #10798: Fix end to end tests for ExplorationEditorImprovementsTab.js (#11619)

* Add waitFor wherever needed

Signed-off-by: prayutsu <gargabhay2000@gmail.com>

* reformat code

* reformat code

* fix lint issue

* Fix suggested changes

* Import waitFor

* Fix suggested changes

nithusha21 commented on scripts/linters/test_files/invalid_test_only.py:
```diff
@@ -0,0 +1,44 @@
+# coding: utf-8
```
Why are these 2 files changed in this PR?

shubhamkorde replied:
I had earlier changed only the three files but some of the checks were failing due to stale build. I tried to fetch upstream and merge it into the branch I created. That led to the changes in these files. How do I correct this?

srijanreddy98 committed:
Fixes Part of #8472: Migrate Exploration data service (#11564)

* Migrate editable exploration backend api service

* Fix types

* Migrate exploration data service

* Refactor to backend api service

* Refactor to backend api service

* Refactor to backend api service

* Refactor to backend api service

* Fix

* Add wait for element to be clickable

* Add a couple more applyAsync

* Migrate exploration-backend-api service

* Migrate exploration-backend-api service

* Add .

* Add .

* Update workflow.js

* Fix exploration diff service

* Update workflow.js

* e2e_Patch 1.1: Fix creator_learner_dashboard_suite
- added waitForAngularEnabled
- added waitForElementToBeClickable
- Fixed null index problem

* e2e_Patch 1.1.1 Fix Minor Linting Issues

* remove waitForAngularEnabled from ExplorationPlayerPage.rateExploration

* fixed clickable button

* Remove wait for angular calls

* fix saving flow issues (also fixes the infamous 'Lost Changes' e2e flake)

* prevent changes when autosave is happening

* fix category input not visible error

* Revert of last two commits (relative to this commit)

* Auto save indicator

* Auto save indicator

* Auto save indicator

* Unit test changes

* Fix e2e tests

* waitFor autosave invisibility before any action on the settings page

* move waitForAutosave to action.js

* add some backend logs to debug 'lost changes'

* print change dicts

* fix e2e tests

* club together changes that occur in quick sequence (200ms) into a single autosave draft backend call

* remove unnecessary logging; add few more action.waitForAutosave()

* revert changes to exp_services.py

* do waitForAutosave before navigating to the main tab in the exp editor

Co-authored-by: darksun27 <sbsiddharth@gmail.com>
Co-authored-by: Kevin Thomas <kevintab@tutanota.com>

ReshuKumari committed:
fixing border-bottom (#11567)

Co-authored-by: Kevin Thomas <kevintab@tutanota.com>

ARKEOLOGIST committed:
Fixed minor issue in correctness footer (#11719)

shubhamkorde committed:
Revert "Fix part of #10306: Added async keyword to asynchronous functions that return a promise"

This reverts commit 473014f35e952bf845a0432d85708b5965e5f9df.

shubhamkorde committed:
Revert "updated changes from develop to fix failing issues"

This reverts commit b5d38bfe012fda6d054f4b334a1d0199d6c85218.

shubhamkorde committed:
Fix part of #10306: Added async keyword to asynchronous functions that return a promise

shubhamkorde committed:
Merge branch 'async-functions-update' of https://github.com/shubhamkorde/oppia into async-functions-update
